src/main/java/net/snowflake/client/core/SFStatement.java

```diff
@@ -123,6 +123,14 @@ private SFBaseResultSet executeQuery(
 
     // snowflake specific client side commands
     if (isFileTransfer(trimmedSql)) {
+      if ((session != null && !session.getJdbcEnablePutGet()) // Server side value
```

Comment:
can refactor to `(session!= null &&!(session.getJdbcEnablePutGet() && session.getEnablePutGet())`

Reply:
It like to keep it simple to understand either of them false then disable it. Sometime negative condition hard to understand. But if you insist, I can make the change.

Reply:
Done.